src/Data/Cache.hs

```diff
@@ -227,11 +227,11 @@ keys = atomically . keysSTM
 now :: IO TimeSpec
 now = getTime Monotonic
 
--- | keeps elements that satify a predicate (used for cache invalidation).
-filterWith :: (Eq k, Hashable k) => (k -> Bool) -> Cache k v -> IO ()
-filterWith f c = atomically $ writeTVar v =<< (HM.filterWithKey (\k _ -> f k) <$> readTVar v) where v = container c
+-- | Keeps elements that satify a predicate (used for cache invalidation).
+filterWithKey :: (Eq k, Hashable k) => (k -> CacheItem v -> Bool) -> Cache k v -> IO ()
```

Comment:
Could we change the signature to `(k -> v -> Bool) -> Cache k v -> IO ()`?

I rather not expose the `CacheItem` type from the public module (as it is internal)

Reply:
Depending on what you want and how you implement it, could you add one of these lines to the docs?

- Note that the predicate might be called for expired items.
- Note that the predicate will not be called for expired items, these will automatically be deleted.